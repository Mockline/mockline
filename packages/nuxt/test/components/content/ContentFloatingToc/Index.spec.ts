import { describe, it, expect } from 'vitest'
import type {TypeOf} from "zod";
// @ts-ignore
import { MContentFloatingToc } from '#components'
import ComponentRender from "../../../component-render";

describe('ContentFloatingToc', () => {
  it.each([
    ['basic case', { props: { links: [{id: 'id', text: 'text', depth: 1}] }}],
    ['<MContentFloatingToc :links="[{id: \'id\', text: \'text\', depth: 1}]"/>'],
    // @ts-expect-error
  ])('renders %s correctly', async (nameOrHtml, options: TypeOf<typeof MContentFloatingToc.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MContentFloatingToc)
    expect(html).toMatchSnapshot()
  })
})


