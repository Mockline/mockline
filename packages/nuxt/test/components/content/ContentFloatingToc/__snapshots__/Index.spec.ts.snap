// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentFloatingToc > renders <MContentFloatingToc :links="[{id: 'id', text: 'text', depth: 1}]"/> correctly 1`] = `"<mcontentfloatingtoc links="[object Object]"></mcontentfloatingtoc>"`;

exports[`ContentFloatingToc > renders basic case correctly 1`] = `
"<div class="fixed z-50 scale-[0.9] sm:scale-[0.6] transition-all duration-300 ease-in-out right-2 sm:right-6 top-1/2 -translate-y-1/2 origin-right hover:scale-110">
  <div class="rounded-md mx-auto transition-all duration-300 ease-in-out p-0 border-transparent">
    <nav class="overflow-y-auto cursor-pointer">
      <div>
        <div>
          <transition-stub enteractiveclass="transition-all duration-100 ease-in-out" enterfromclass="opacity-0" entertoclass="opacity-100" leaveactiveclass="transition-all duration-100 ease-in-out" leavefromclass="opacity-100" leavetoclass="opacity-0" mode="out-in" appear="false" persisted="false" css="true">
            <div class="space-y-4">
              <div class="flex flex-col items-end gap-1">
                <div class="h-[4px] transition-all duration-500 ease-in-out bg-[var(--color-bg-quaternary)] w-12 cursor-pointer"></div>
                <!--v-if-->
              </div>
            </div>
          </transition-stub>
        </div>
      </div>
    </nav>
  </div>
</div>"
`;

exports[`Index > can mount component 1`] = `
"<div id="test-wrapper">
  <div class="fixed z-50 scale-[0.9] sm:scale-[0.6] transition-all duration-300 ease-in-out right-2 sm:right-6 top-1/2 -translate-y-1/2 origin-right hover:scale-110">
    <div class="rounded-md mx-auto transition-all duration-300 ease-in-out p-0 border-transparent">
      <nav class="overflow-y-auto cursor-pointer">
        <div>
          <div>
            <transition-stub enteractiveclass="transition-all duration-100 ease-in-out" enterfromclass="opacity-0" entertoclass="opacity-100" leaveactiveclass="transition-all duration-100 ease-in-out" leavefromclass="opacity-100" leavetoclass="opacity-0" mode="out-in" appear="false" persisted="false" css="true">
              <!--v-if-->
            </transition-stub>
          </div>
        </div>
      </nav>
    </div>
  </div>
</div>"
`;
